app/controllers/campaign_pages_controller.rb

```diff
@@ -15,8 +15,9 @@ def new
 
   def create
     @campaign_page = CampaignPage.new(@page_params)
-    @campaign_page.compile_html
+    return switch_template(@campaign_page, :new) if params[:switch_template]
```

Comment:
What if the API looked something like this:

```ruby
def update
  page_service = CampaignPageMaker.new(page_params) #or some other name

 if page_service.switched_template?
   render page_service.view
 else
   redirect_to page_service.campaign_page
 end
end
```

Similar for `create`.

Sorry, I rushed this... it's a quick idea for how it could look. If you're not happy, then let's talk about it on Monday.